gen/map.go

```diff
@@ -281,3 +281,144 @@ func (m *mapGenerator) equalsUnhashable(g Generator, spec *compile.MapSpec) (str
 
 	return name, wrapGenerateError(spec.ThriftName(), err)
 }
+
+func (m *mapGenerator) zapMarshaler(
+	g Generator,
+	spec compile.TypeSpec,
+	root *compile.MapSpec,
+	fieldValue string,
+) (string, error) {
+	name := zapperName(g, spec)
+	switch root.KeySpec.(type) {
+	case *compile.StringSpec:
+		// We are already at the root, so no need to check if it is a typedef of a
+		// string. For simplicity, we always cast the key to a string when logging this way.
+		if err := g.EnsureDeclared(
```

Comment:
You're dealing with the root `map[...]` type but the key could still be a typedef of a string so you still want inspect `RootTypeSpec(root.KeySpec)`.

```
typedef string Foo
typedef map<Foo, i32> Bar
```